packages/teams-js/test/public/authentication.spec.ts

```diff
@@ -25,98 +32,74 @@ describe('authentication', () => {
   });
 
   it('should not allow authentication.authenticate calls before initialization', () => {
+    expect.assertions(1);
     const authenticationParams: authentication.AuthenticatePopUpParameters = {
       url: 'https://someurl/',
       width: 100,
       height: 200,
     };
 
-    expect(() => authentication.authenticate(authenticationParams)).toThrowError(
+    return expect(() => authentication.authenticate(authenticationParams)).toThrowError(
       'The library has not yet been initialized',
     );
   });
 
-  it('should not allow authentication.authenticate calls from authentication context', () => {
-    return utils.initializeWithContext('authentication').then(() => {
-      const authenticationParams: authentication.AuthenticatePopUpParameters = {
-        url: 'https://someurl/',
-        width: 100,
-        height: 200,
-      };
-
-      expect(() => authentication.authenticate(authenticationParams)).toThrowError(
-        'This call is only allowed in following contexts: ["content","sidePanel","settings","remove","task","stage","meetingStage"]. Current context: "authentication".',
-      );
+  const allowedContexts = [
+    FrameContexts.content,
+    FrameContexts.sidePanel,
+    FrameContexts.settings,
+    FrameContexts.remove,
+    FrameContexts.task,
+    FrameContexts.stage,
+    FrameContexts.meetingStage,
+  ];
+
+  Object.keys(FrameContexts)
+    .map(k => FrameContexts[k])
+    .forEach(context => {
+      if (allowedContexts.some(allowedContext => allowedContext === context)) {
+        it(`should allow authentication.authenticate calls from ${context} context`, async () => {
+          expect.assertions(1);
+          await utils.initializeWithContext(context);
+
+          const authenticationParams: authentication.AuthenticatePopUpParameters = {
+            url: 'https://someurl/',
+            width: 100,
+            height: 200,
+          };
+          const promise = authentication.authenticate(authenticationParams);
+
+          utils.processMessage({
+            origin: utils.tabOrigin,
+            source: utils.childWindow,
+            data: {
+              id: 0,
+              func: 'authentication.authenticate.success',
+              args: [mockResult],
+            },
+          } as MessageEvent);
+
+          return expect(promise).resolves.toEqual(mockResult);
+        });
+      } else {
+        it(`should not allow authentication.authenticate calls from ${context} context`, async () => {
+          expect.assertions(1);
+          await utils.initializeWithContext(context);
+          const authenticationParams: authentication.AuthenticatePopUpParameters = {
+            url: 'https://someurl/',
+            width: 100,
+            height: 200,
+          };
+
+          return expect(() => authentication.authenticate(authenticationParams)).toThrowError(
```

Comment:
Can we use await instead of return here?

Reply:
Just removed the return because this isn't an async function